src/Microsoft.Health.Fhir.Azure/ContainerRegistry/ContainerRegistryBasicTokenProvider.cs

```diff
@@ -0,0 +1,62 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using EnsureThat;
+using Microsoft.Extensions.Options;
+using Microsoft.Health.Fhir.Core.Configs;
+using Microsoft.Health.Fhir.Core.Features.Operations.DataConvert;
+using Microsoft.Health.Fhir.Core.Features.Operations.DataConvert.Models;
+
+namespace Microsoft.Health.Fhir.Azure.ContainerRegistry
+{
+    /// <summary>
+    /// Implement basic auth logic using ACR admin keys.
+    /// Reference: https://github.com/Azure/acr/blob/main/docs/AAD-OAuth.md#catalog-listing-using-spadmin-with-basic-auth
+    /// </summary>
+    public class ContainerRegistryBasicTokenProvider : IContainerRegistryTokenProvider
+    {
+        private readonly DataConvertConfiguration _dataConvertConfiguration;
+
+        public ContainerRegistryBasicTokenProvider(IOptions<DataConvertConfiguration> dataConvertConfiguration)
+        {
+            EnsureArg.IsNotNull(dataConvertConfiguration, nameof(dataConvertConfiguration));
+
+            _dataConvertConfiguration = dataConvertConfiguration.Value;
+        }
+
+#pragma warning disable CS1998 // Async method lacks 'await' operators and will run synchronously
+        public async Task<string> GetTokenAsync(string registryServer, CancellationToken cancellationToken)
+#pragma warning restore CS1998 // Async method lacks 'await' operators and will run synchronously
+        {
+            var containerRegistryInfo = _dataConvertConfiguration.ContainerRegistries
+                .FirstOrDefault(registry => string.Equals(registry.ContainerRegistryServer, registryServer, StringComparison.OrdinalIgnoreCase));
+            if (containerRegistryInfo == null)
+            {
+                throw new ContainerRegistryNotRegisteredException(string.Format(Resources.ContainerRegistryNotRegistered, registryServer));
+            }
+
+            if (string.IsNullOrEmpty(containerRegistryInfo.ContainerRegistryServer)
+                || string.IsNullOrEmpty(containerRegistryInfo.ContainerRegistryUsername)
+                || string.IsNullOrEmpty(containerRegistryInfo.ContainerRegistryPassword))
+            {
+                throw new ContainerRegistryNotAuthorizedException(string.Format(Resources.ContainerRegistryNotAuthorized, containerRegistryInfo.ContainerRegistryServer));
```

Comment:
>ContainerRegistryNotAuthorized [](start = 90, length = 30)

nit: Authorization means being allowed access to the system, while here we have case for misconfiguration rather than you not allowed.
I also think this should return 500, it's not client fault what something is misconfigured on server side.

Reply:
Agree. I will change the exception & description to configuration error and set response as internal error. Besides, since these misconfigurations could be very likely to happen, will this 500 error code trigger any alerts when we ship this feature to PaaS service?

Reply:
I don't know much about PaaS part, so maybe @LTA-Thinking or @feordin  can help you with this question?

Reply:
Discussed with the PaaS team, 500 for unexpected errors (not triggered by user actions like acr down, network down) is acceptable.